### ISSUE TITLE
New Swift 4.2 complains about Usabilla 5.0.2

### ISSUE THREAD

**Issue by StammTom** [usabilla/usabilla-u4a-ios-swift-sdk#109]
The newly release XCode10 says "Module compiled with Swift 4.1.2 cannot be imported by Swift 4.2 compiler."

**Comment by TranTo-To** [comment]
yep. Look forward to having the framework compiled with swift 4.2 soon 
![screen shot 2018-09-18 at 11 35 00 am](https://user-images.githubusercontent.com/32047496/45659268-f0f4c000-bb36-11e8-94bf-dea7c221817f.png)

**Comment by Rifinio** [comment]
Hello @StammTom, @TranTo-To 
Did you use the latest [release](https://github.com/usabilla/usabilla-u4a-ios-swift-sdk/releases/tag/v5.0.2-Xcode-10-beta6). It is compiled using swift 4.2.

**Comment by TranTo-To** [comment]
Yes, I did. And Apple released Xcode 10 officially recently. So the latest release gave me that error result. @Rifinio you may want to recompile it

**Comment by Rifinio** [comment]
Thank you @alessandroboron,
We will take a look asap to this and communicate back on this issue when it is done!

**Comment by mariohahn** [comment]
Pls update your Pod

**Comment by StammTom** [comment]
> Hello @StammTom, @TranTo-To
> Did you use the latest [release](https://github.com/usabilla/usabilla-u4a-ios-swift-sdk/releases/tag/v5.0.2-Xcode-10-beta6). It is compiled using swift 4.2.

Xcode10 and Swift4.2 download yesterday (iOS12 day).  No betas

**Comment by Rifinio** [comment]
Hello guys,
A new release is out and you can find under the tag [v5.0.2-Xcode10-GM](https://github.com/usabilla/usabilla-u4a-ios-swift-sdk/releases/tag/v5.0.2-Xcode10-GM).
Thank you for your patience!

**Comment by mariohahn** [comment]
There is still no Podspec update. 🤨

**Comment by alessandroboron** [comment]
There’s an issue with Carthage. It says that the project has not shared schemes.

**Comment by koenpunt** [comment]
> Hello guys,
> A new release is out and you can find under the tag [v5.0.2-Xcode10-GM](https://github.com/usabilla/usabilla-u4a-ios-swift-sdk/releases/tag/v5.0.2-Xcode10-GM).
> Thank you for your patience!

Please tag as a final release (so without the `-Xcode10-GM` suffic), because Xcode 10 is also final

**Comment by Rifinio** [comment]
@alessandroboron we tested the last tagged release in a new project using Xcode 10 GM and all worked fine! 
using this line in the Cartfile `github "usabilla/usabilla-u4a-ios-swift-sdk" "v5.0.2-Xcode10-GM"`
And for Cocoapods using :
`pod 'Usabilla', :git => 'https://github.com/usabilla/usabilla-u4a-ios-swift-sdk.git', :tag => 'v5.0.2-Xcode10-GM'`

**Comment by mariohahn** [comment]
I don't want to refer to a special tag.  Why can't you just update your main pod spec and push it to cocoapods?
 
It's kind of frustrating to use your SDK. 😕

**Comment by plezervi** [comment]
I am unable to build my project with fastlane, using the v5.0.2-Xcode10-GM version.

```
-= Output =-
Undefined symbols for architecture armv7:
"___llvm_profile_runtime", referenced from:
__hidden#452_ in 001.o
__hidden#452_ in 002.o
__hidden#452_ in 003.o
__hidden#452_ in 004.o
__hidden#452_ in 005.o
__hidden#452_ in 006.o
__hidden#452_ in 007.o
...
ld: symbol(s) not found for architecture armv7
Exited with 1


error: Failed to compile bundle: /var/folders/vq/j6j2lh1s3cz7cbpw7shc9149fy4t4_/T/UsabillavOZMeH/Usabilla.armv7.xar
```

**Comment by alessandroboron** [comment]
@Rifinio my bad I was pointing to **v5.0.2-Xcode-10-GM** which apparently is an old one instead of pointing to **v5.0.2-Xcode10-GM**. Btw XCode is officially released so you guys my want to change the name of the tag. Cheers

**Comment by RobertoRiquelme** [comment]
How about using **v5.0.2-Xcode-10-GM** as an **s.dependency**
It is not possible to add this git/branch to a podspec.
Please! add it to your main podspec! 🙏

**Comment by hkramer70** [comment]
Some news about Xcode10 and Swift4.2 (via carthage distribution)? We are waiting now for several days and I won't start praying at you to get a valid version. In the worst case we have to look for other libs.

**Comment by StammTom** [comment]
@abrljak:  Yikes!  We have a release "Real-soon-now" .  If this is a problem, it is a problem for us too!

**Comment by tzengerink** [comment]
We've just pushed version [v5.1.0](https://github.com/usabilla/usabilla-u4a-ios-swift-sdk/releases/tag/v5.1.0) that is compatible with XCode10. With this version there is no longer a need to use a special tag, the podspec has been updated and the release should be properly usable via Carthage.

Thanks for the patience.

**Comment by oanhof** [comment]
So I just updated to the new 5.1.0 version and uploaded a build to Apple. Unfortunately I got the following message from Apple:

> Invalid Bundle - Disallowed LLVM instrumentation. Do not submit apps with LLVM profiling instrumentation or coverage collection enabled. Turn off LLVM profiling or code coverage, rebuild your app and resubmit the app.

**Comment by ctreffs** [comment]
I am experiencing similar problems while enterprise signing with the version 5.1.0 and Xcode 10.
We sign on the ci.

**Comment by copa** [comment]
... same here . We're using fastlane and jenkins.

**Comment by ctreffs** [comment]
Could fix my problem, which was an issue of the `bitcode-build-tool` triggered through fastlane's gym:

Add these to your Gymfile
```ruby
include_bitcode false

export_options(
  uploadBitcode: false,
  compileBitcode: false
)
```

Maybe this helps others

**Comment by copa** [comment]
@ctreffs your fix worked here too:

Alternatively use this syntax: 
 gym(
     ....
      export_options: {
           uploadBitcode: false,
           compileBitcode: false
         },
       include_bitcode: false
    )

**Comment by StammTom** [comment]
@tzengerink does this mean the LLVM issue is fixed?

**Comment by copa** [comment]
@tzengerink If your project uses the swift 4.2 compiler, I'd say yes. Disabling bitcode isn't nice, especially because a dependency forces us to do. Anyhow, for now we have no other choice.

**Comment by mthormann-dhi** [comment]
@oanhof I spun up a [new issue](https://github.com/usabilla/usabilla-u4a-ios-swift-sdk/issues/113) specifically for this on Friday. Ran into it as well and verified it was Usabilla. I'm  guessing someone left code coverage or something turned on in the release build by accident.

> So I just updated to the new 5.1.0 version and uploaded a build to Apple. Unfortunately I got the following message from Apple:
> 
> > Invalid Bundle - Disallowed LLVM instrumentation. Do not submit apps with LLVM profiling instrumentation or coverage collection enabled. Turn off LLVM profiling or code coverage, rebuild your app and resubmit the app.
> 
> This must be due to Usabilla because I was able to upload builds just fine before I included the SDK in my app.

**Comment by mthormann-dhi** [comment]
@oanhof v5.1.2 fixes the issue

**Comment by torkvell** [comment]
Hi @babuiglesiaskhan,

Thanks for reaching out to support!

Could you please tell me a bit more about your setup? Are you using 32-bit modules? Do note that 32-bit modules are not supported in the 5.1.4 version, but will be supported in our next major release that will happen sometime within the next week.

Looking forward to your reply.

**Comment by babuiglesiaskhan** [comment]
Hi @torkvell ,
I don't know about 32-bit modules. Can you tell me how can I check it?

**Comment by torkvell** [comment]
@babuiglesiaskhan 

These are the steps:
1. Select your project
2. Select your target
3. Select Build Settings
4. Look for Valid Architectures
